datasets/covid19_tracking/state_screenshots/custom/download_screenshots.py

```diff
@@ -17,42 +17,55 @@
 import os
 import pathlib
 import typing
+from multiprocessing.pool import ThreadPool
 
 import requests
-from google.cloud import storage
 
 
-def main(csv_path: pathlib.Path, source_column: str, target_column: str):
-    gcs_client = storage.Client()
+def download_item(source_target: typing.Tuple[str, pathlib.Path]):
+    source_url, download_path = source_target
+    if not os.path.exists(download_path):
+        r = requests.get(source_url, stream=True)
+        if r.status_code == 200:
+            with open(download_path, "wb") as f:
+                for chunk in r:
+                    f.write(chunk)
+
+
+def download_parallel(source_targets: typing.List[typing.Tuple[str, pathlib.Path]]):
+    ThreadPool(10).imap_unordered(download_item, source_targets)
+
+
+def main(csv_path: pathlib.Path, source_column: str, download_prefix: str):
     with open(csv_path) as csv_file:
         csv_reader = csv.DictReader(csv_file, delimiter=",")
+        download_dir = pathlib.Path(download_prefix)
+
+        row_num = 0
+        source_targets = []
         for row in csv_reader:
+            # Example:
+            # https://covidtracking.com/screenshots/AL/AL-20210307-230802.png
             source_url = row[source_column]
-            gcs_bucket, gcs_path = decompose_gcs_uri(row[target_column])
-
-            stream_http_to_gcs(gcs_client, source_url, gcs_bucket, gcs_path)
-
+            state, filename = row[source_column].split("/")[-2:]
 
-def decompose_gcs_uri(gcs_uri: str) -> typing.Tuple[str, str]:
-    gcs_bucket, gcs_path = gcs_uri.replace("gs://", "").split("/", 1)
-    return gcs_bucket, gcs_path
+            (download_dir / state).mkdir(parents=True, exist_ok=True)
+            source_targets.append((source_url, download_dir / state / filename))
 
+            row_num += 1
+            if row_num % 100 == 0:
```

Comment:
100 is also a magic number. Probably we want a constant which is a multiple of the number of threads?

Reply:
I removed that in favour of using `WORKER_THREADS`, so that we're chunking based on how many threads we have for simplicity: one item processed per thread per call to `imap_unordered`.

Please check edb409f.